src/Text/Pandoc/Writers/Txt2Tags.hs

```diff
@@ -0,0 +1,512 @@
+consolidateRawBlocks (x:xs) = x : consolidateRawBlocks xs
+
+-- | Convert list of Pandoc inline elements to Txt2Tags.
+inlineListToTxt2Tags :: PandocMonad m
+                     => WriterOptions -> [Inline] -> Txt2Tags m Text
+inlineListToTxt2Tags opts lst =
+  T.concat <$> mapM (inlineToTxt2Tags opts) lst
+
+-- | Convert Pandoc inline element to Txt2Tags.
+inlineToTxt2Tags :: PandocMonad m
+                 => WriterOptions -> Inline -> Txt2Tags m Text
+
+inlineToTxt2Tags opts (Span _attrs ils) =
+  inlineListToTxt2Tags opts ils
+
+inlineToTxt2Tags opts (Emph lst) = do
+  contents <- inlineListToTxt2Tags opts lst
+  return $ "//" <> contents <> "//"
+
+inlineToTxt2Tags opts (Underline lst) = do
+  contents <- inlineListToTxt2Tags opts lst
+  return $ "__" <> contents <> "__"
+
+inlineToTxt2Tags opts (Strong lst) = do
+  contents <- inlineListToTxt2Tags opts lst
+  return $ "**" <> contents <> "**"
+
+inlineToTxt2Tags opts (Strikeout lst) = do
+  contents <- inlineListToTxt2Tags opts lst
+  return $ "--" <> contents <> "--"
+
+inlineToTxt2Tags opts (Superscript lst) = do
+  contents <- inlineListToTxt2Tags opts lst
+  return $ "<sup>" <> contents <> "</sup>"
+
+inlineToTxt2Tags opts (Subscript lst) = do
+  contents <- inlineListToTxt2Tags opts lst
+  return $ "<sub>" <> contents <> "</sub>"
+
+inlineToTxt2Tags opts (SmallCaps lst) = inlineListToTxt2Tags opts lst
+
+inlineToTxt2Tags opts (Quoted SingleQuote lst) = do
+  contents <- inlineListToTxt2Tags opts lst
+  return $ "\8216" <> contents <> "\8217"
+
+inlineToTxt2Tags opts (Quoted DoubleQuote lst) = do
+  contents <- inlineListToTxt2Tags opts lst
+  return $ "\8220" <> contents <> "\8221"
+
+inlineToTxt2Tags opts (Cite _  lst) = inlineListToTxt2Tags opts lst
+
+inlineToTxt2Tags _ (Code _ str) =
+  -- In Txt2Tags, text surrounded by '' is really just a font statement, i.e. <tt>,
+  -- and so other formatting can be present inside.
+  -- However, in pandoc, and markdown, inlined code doesn't contain formatting.
+  -- So I have opted for using %% to disable all formatting inside inline code blocks.
+  -- This gives the best results when converting from other formats to Txt2Tags, even if
+  -- the resultand code is a little ugly, for short strings that don't contain formatting
+  -- characters.
+  -- It does mean that if pandoc could ever read Txt2Tags, and so round-trip the format,
+  -- any formatting inside inlined code blocks would be lost, or presented incorrectly.
+  return $ "''%%" <> str <> "%%''"
+
+inlineToTxt2Tags _ (Str str) = return $ escapeString str
+
+inlineToTxt2Tags _ (Math mathType str) = return $ delim <> str <> delim
+                                 -- note:  str should NOT be escaped
+  where delim = case mathType of
+                     DisplayMath -> "$$"
+                     InlineMath  -> "$"
+
+inlineToTxt2Tags _ il@(RawInline f str)
+  | f == Format "Txt2Tags" = return str
+  | f == Format "html"     = return $ "<html>" <> str <> "</html>"
```

Comment:
I suppose there is no difference. Again, this part is straight from the DokuWiki.hs file: 

![image](https://user-images.githubusercontent.com/477243/104966034-43349d00-59e0-11eb-9f3d-af0aef3ce9da.png)


> Those are good news, thanks! I went through the code and it looks good; I made inline comments whenever I was unsure about something.
> 
> I'm not sure if @jgm considers it essential, but it would be best if we could switch the result type of `blockToTxt2Tags` from `Text` to `Doc Text` so pandoc's `--columns` option would work better with this writer. Do you have time to give it a try?

I don't know how to do this, sorry. I've just hacked the existing dokuwiki code so it's outputting valid txt2tags markup (most of the difference is with the headings)